src/offer.js

```diff
@@ -44,9 +45,17 @@ function getOrganisation(result, grouped, orgUrl) {
   }
 
   return request.then(response => {
-    result.logo = response.data.logo || result.logo;
-    result.primary_color = response.data.primary_color || result.primary_color;
-    result.secondary_color = response.data.secondary_color || result.secondary_color;
+    const data = response.data;
+
+    result.logo = data.logo || result.logo;
+    result.primary_color = data.primary_color || result.primary_color;
+    result.secondary_color = data.secondary_color || result.secondary_color;
+
+    if (_get(data, 'payment.source_id_type') === sourceIdType && _get(data, 'payment.url')) {
+      result.paymentUrl = data.payment.url.replace('{source_id}', sourceId)
+                                          .replace('{offer_id}', result.id);
```

Comment:
I've also been using a regular expression to check for replacement strings e.g:
link.replace(/{source_id}/g, data.source_id);
as this will find and replace multiple occurrences of source_id in the string (just in case somebody really wanted to do that!)